### DISCUSSION TITLE
/.well-known/jwk.pem output is incorrect

### DISCUSSION THREAD

**Discussion by gcleaves** [requarks/wiki#6328] [Error / Bug Report] [answered]
The string produced by /.well-known/jwk.pem does not represent a valid PEM key. 

In order to generate a valid PEM key it appears necessary to feed the output of /.well-known/jwk.json into a JWK to PEM converter suck as https://8gwifi.org/jwkconvertfunctions.jsp.

**Comment by NGPixel** [comment]
Are you using the raw output of the page and not the formatted one by the browser (which removes the line breaks?).

The pem public key is valid and is the output of the node crypto generateKeyPairSync method (https://github.com/requarks/wiki/blob/main/server/core/auth.js#L367-L385)

**Comment by gcleaves** [comment]
Hi. I added line breaks to the browser formatted output. 

I'll post the "evidence" when I return from the long weekend.

It now strikes me it might be an encoding issue that depends on the
key? Maybe some unsafe characters where translated by my browser. Anyway,
I'll post my examples later.

It sounds like the problem is on my end, thanks for your response.

**Reply by NGPixel** [reply]
Well, you have to use the raw output... It's no wonder it doesn't work otherwise.

**Comment by gcleaves** [comment, marked as answer]
I see the actual issue. Wiki.js outputs a key starting with "BEGIN RSA PUBLIC KEY" and encoded accordingly while https://token.dev and the Lua JWT verification library I'm using expect a key starting with "BEGIN PUBLIC KEY" and encoded differently.

Here an example of the difference I'm speaking of: https://runkit.com/gcleaves/wiki-js-pem

It looks like wiki.js uses PEM type pkcs1 while PKCS#8 appears to be more common, at least from the two "libraries" I used.
